modules/custom/alternative_frontpage/src/EventSubscriber/RedirectHomepageSubscriber.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+namespace Drupal\alternative_frontpage\EventSubscriber;
+
+use Drupal\Core\Path\PathMatcher;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\EventDispatcher\Event;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpKernel\KernelEvents;
+use Drupal\user\UserData;
+use Drupal\Core\Session\AccountProxy;
+use Drupal\Core\Config\ConfigFactory;
+
+/**
+ * Class RedirectHomepageSubscriber.
+ */
+class RedirectHomepageSubscriber implements EventSubscriberInterface {
+
+  /**
+   * Protected var UserData.
+   *
+   * @var \Drupal\user\UserData
+   */
+  protected $userData;
+
+  /**
+   * Protected var ConfigFactory.
+   *
+   * @var \Drupal\Core\Config\ConfigFactory
+   */
+  protected $configFactory;
+
+  /**
+   * Protected var for the current user.
+   *
+   * @var \Drupal\Core\Session\AccountProxy
+   */
+  protected $currentUser;
+
+  /**
+   * Protected var for the path matcher.
+   *
+   * @var \Drupal\Core\Path\PathMatcher
+   */
+  protected $pathMatcher;
+
+  /**
+   * Constructor for the RedirectHomepageSubscriber.
+   */
+  public function __construct(UserData $user_data, ConfigFactory $config_factory, AccountProxy $current_user, PathMatcher $path_matcher) {
+    // We needs it.
+    $this->userData = $user_data;
+    $this->configFactory = $config_factory->get('alternative_frontpage.settings');
+    $this->currentUser = $current_user;
+    $this->pathMatcher = $path_matcher;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function getSubscribedEvents() {
+    // 280 priority is higher than the dynamic and static page cache.
```

Comment:
I don't really understand where this 280 comes from or is set too. Are we sure it is now in the correct order?